Examples/ComplexVaporExample/Sources/App/Kitten.swift

```diff
@@ -0,0 +1,34 @@
+import Foundation
+import MongoSwift
+import Vapor
+
+/// Possible cat food choices.
+enum CatFood: String, Codable {
+    case salmon,
+        tuna,
+        chicken,
+        turkey,
+        beef
+}
+
+/// The structure of a food update request.
+struct FoodUpdate: Codable {
+    let favoriteFood: CatFood
+}
+
+/// Represents a kitten.
+struct Kitten: Content {
```

Comment:
initially I had a birthdate `Date` field here. however, it seems that somewhere in the Leaf 4 development process they removed support for the `#date()` formatter tag and have not added it back yet, so there was not an easy way for me to turn the date number representation into something human-readable without using some hacky Javascript or something on the page.
Leaf also does not use `JSONEncoder` to encode the provided context, it uses this custom encoder `LeafEncoder`, so I couldn't just choose a custom encoding strategy for dates to use there.

anyway I think this is interesting enough of an example type that we're probably fine, but if you think there's something else I should add in place of birthdate let me know.